bundles/org.openhab.binding.smaenergymeter/src/main/java/org/openhab/binding/smaenergymeter/internal/discovery/SMAEnergyMeterDiscoveryService.java

```diff
@@ -54,35 +62,52 @@ public Set<ThingTypeUID> getSupportedThingTypes() {
 
     @Override
     protected void startBackgroundDiscovery() {
+        if (packetListener != null) {
+            return;
+        }
+
         logger.debug("Start SMAEnergyMeter background discovery");
-        scheduler.schedule(this::discover, 0, TimeUnit.SECONDS);
+        try {
+            packetListener = listenerRegistry.getListener(PacketListener.DEFAULT_MCAST_GRP,
+                    PacketListener.DEFAULT_MCAST_PORT);
+            packetListener.open();
+        } catch (IOException e) {
+            logger.error("Could not start background discovery", e);
+            return;
+        }
+
+        packetListener.addPayloadHandler(this);
     }
 
     @Override
-    public void startScan() {
-        logger.debug("Start SMAEnergyMeter scan");
-        discover();
+    protected void stopBackgroundDiscovery() {
+        packetListener.removePayloadHandler(this);
     }
 
-    private synchronized void discover() {
-        logger.debug("Try to discover a SMA Energy Meter device");
-
-        EnergyMeter energyMeter = new EnergyMeter(EnergyMeter.DEFAULT_MCAST_GRP, EnergyMeter.DEFAULT_MCAST_PORT);
+    @Override
+    public void startScan() {
+        logger.debug("Start SMAEnergyMeter scan");
+        startBackgroundDiscovery();
         try {
-            energyMeter.update();
-        } catch (IOException e) {
-            logger.debug("No SMA Energy Meter found.");
-            logger.debug("Diagnostic: ", e);
-            return;
+            Thread.sleep(60_000);
+        } catch (InterruptedException e) {
+            logger.debug("Discovery task terminated", e);
+        } finally {
+            stopBackgroundDiscovery();
         }
+    }
 
-        logger.debug("Adding a new SMA Engergy Meter with S/N '{}' to inbox", energyMeter.getSerialNumber());
+    @Override
+    public void handle(EnergyMeter energyMeter) throws IOException {
+        String identifier = energyMeter.getSerialNumber();
+        logger.debug("Adding a new SMA Energy Meter with S/N '{}' to inbox", identifier);
         Map<String, Object> properties = new HashMap<>();
         properties.put(Thing.PROPERTY_VENDOR, "SMA");
-        properties.put(Thing.PROPERTY_SERIAL_NUMBER, energyMeter.getSerialNumber());
-        ThingUID uid = new ThingUID(THING_TYPE_ENERGY_METER, energyMeter.getSerialNumber());
+        properties.put(Thing.PROPERTY_SERIAL_NUMBER, identifier);
+        ThingUID uid = new ThingUID(THING_TYPE_ENERGY_METER, identifier);
```

Comment:
Identifier is a number turned into String so additional formatting is not needed. If device brings invalid characters for identifier I'd rather let framework mark thing as invalid instead of  accepting it.